src/Symfony/Component/Messenger/Bridge/MongoDB/Transport/Receiver.php

```diff
@@ -0,0 +1,151 @@
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\MongoDB\Transport;
+
+use MongoDB\BSON\ObjectId;
+use MongoDB\Model\BSONDocument;
+use Symfony\Component\Messenger\Bridge\MongoDB\Stamp\ReceivedStamp;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\MessageDecodingFailedException;
+use Symfony\Component\Messenger\Stamp\TransportMessageIdStamp;
+use Symfony\Component\Messenger\Transport\Receiver\ListableReceiverInterface;
+use Symfony\Component\Messenger\Transport\Receiver\MessageCountAwareInterface;
+use Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+final class Receiver implements ReceiverInterface, MessageCountAwareInterface, ListableReceiverInterface
+{
+    public function __construct(
+        private readonly Connection $connection,
+        private readonly SerializerInterface $serializer,
+    ) {
+    }
+
+    /**
+     * @return array{0?: Envelope}
+     */
+    public function get(): array
+    {
+        $document = $this->connection->get();
+
+        if (!$document instanceof BSONDocument) {
+            return [];
+        }
+
+        return [$this->createEnvelope($document)];
+    }
+
+    public function ack(Envelope $envelope): void
+    {
+        $stamp = $envelope->last(ReceivedStamp::class);
+
+        if (!$stamp instanceof ReceivedStamp) {
+            throw new \LogicException('Unable to retrieve ReceivedStamp on the envelope');
+        }
+
+        $this->connection->ack($stamp->getId());
+    }
+
+    public function reject(Envelope $envelope): void
+    {
+        $stamp = $envelope->last(ReceivedStamp::class);
+
+        if (!$stamp instanceof ReceivedStamp) {
+            throw new \LogicException('Unable to retrieve ReceivedStamp on the envelope');
+        }
+
+        $this->connection->reject($stamp->getId());
+    }
+
+    /**
+     * @return \Generator<Envelope>
+     */
+    public function all(?int $limit = null): iterable
+    {
+        foreach ($this->connection->findAll($limit) as $document) {
+            yield $this->createEnvelope($document);
+        }
+    }
+
+    /**
+     * @param array<string, mixed>|object $filters
+     * @param array<string, mixed>        $options
+     *
+     * @return \Generator<Envelope>
+     */
+    public function findBy($filters, array $options): \Generator
+    {
+        foreach ($this->connection->findBy($filters, $options) as $document) {
+            yield $this->createEnvelope($document);
+        }
+    }
+
+    /**
+     * @param array<string, mixed>|object $filters
+     * @param array<string, mixed>        $options
+     */
+    public function countBy($filters, array $options): int
```

Comment:
Those 2 methods are not part of the implemented interfaces. I don't think they should be defined here